### PR TITLE
Fix #1257 Manage core menu visibility

### LINKED ISSUE #1257: Add the possibility to show opfab menu to only a set of groups
As for business menu , you can choose to show menu depending of user group membership , do the same for native opfab menu and submenu ( Feed , Archives ......  ) 

The configuration has to be set in ui-menu.json . 

Remove navbar.hidden in web-ui.json 

Possible configuration : 
{
  "coreMenusVisibility":
  [
    {
      "id": "feed",
      "visible": true,
      "showOnlyForGroups": ["ReadOnly","Planner"]
    }
    
  ],

  "menus": [
    {
      "id": "menu1",
      "label": "title.single",
      "entries": [ ......... 

 

visible set to false or true by default ?? → to discuss

i would propose false so when we introduce a new menu , the opfab user shall not see it if does not touch his config (this way we can for example introduce new menu for beta features with no impact an exitisting installation)

 

Modify default config to use this option and add cypress test

### PR DESCRIPTION
**Pull request by AlexGuironnetRTE**
Fix #1257

In release notes, under features:

FE-1257 : Manage core menu visibility
+
Core menus such as the archives, usercard creation, settings etc. are now managed in the ui-menu.json file, and can be made visible to certain groups only just like custom menus. See the migration guide for details as it will require changes to your configuration.
//TODO Link to migration guide

Signed-off-by: Alexandra Guironnet <alexandra.guironnet@rte-france.com>

### CODE REVIEW

**Review by freddidierRTE — CHANGES_REQUESTED** [review]
Solve sonar issue : https://sonarcloud.io/project/issues?id=org.lfenergy.operatorfabric%3Aoperatorfabric-core&pullRequest=1388&resolved=false&types=CODE_SMELL

File: src/docs/asciidoc/resources/migration_guide_to_2.8.adoc [hunk 1]
```diff
@@ -35,7 +35,56 @@ cd src/test/resources
...
 
+The visibility of some core OperatorFabric menus (monitoring, logging, feed configuration screen,...) was so far
+configurable for a given OperatorFabric instance through various properties in `web-ui.json`.
+
+As of this version, it has been unified with that of custom menus:
+
+* It will now be managed in `ui-menu.json` along with custom menus
+* It is now possible to make these menus visible only for certain groups
+
+.Deprecated properties in `web-ui.json`
```
[the remark this answers is not in the document]
Comment (AlexGuironnetRTE) [reply src/docs/asciidoc/resources/migration_guide_to_2.8.adoc:48]: True, I changed the text to make it clear that the properties are not taken into account starting with this version.

File: ui/main/src/app/app.component.ts [hunk 2]
```diff
@@ -90,6 +91,14 @@ export class AppComponent implements OnInit {
...
     }));
 
+    this.configService.loadCoreMenuConfigurations().subscribe(() => {
+          console.log(new Date().toISOString(), `Configuration for core menus loaded (ui-menu.json)`);
+        },
+        catchError((err, caught) => {
+          console.error('Impossible to load configuration file ui-menu.json', err);
+          return caught;
+        }));
+
```
[the remark this answers is not in the document]
Comment (AlexGuironnetRTE) [reply ui/main/src/app/app.component.ts:101]: As discussed, it doesn't really need authentication to be loaded but there is no reason to load it "first" like web-ui.json so I'm moving it to initApplication.
(web-ui needs to be loaded before auth is attempted as it defines the authentication mode).

**Comment by AlexGuironnetRTE** [comment]
> Solve sonar issue : https://sonarcloud.io/project/issues?id=org.lfenergy.operatorfabric%3Aoperatorfabric-core&pullRequest=1388&resolved=false&types=CODE_SMELL

Done

**Pull request merged**